code/game/turfs/open/floor/misc_floor.dm

```diff
@@ -190,3 +190,122 @@
 	base_icon_state = "tcomms"
 	icon = 'icons/turf/floors/misc.dmi'
 	color = null
+
+//ship turfs
+/turf/open/floor/ship
+	name = "Ship Plating"
+	desc = "Report a bug if you see this."
+
+/turf/open/floor/ship/dirt
+	gender = PLURAL
+	name = "dirt"
+	desc = "Upon closer examination, it's still dirt."
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "dirt"
+	footstep = FOOTSTEP_SAND
+	barefootstep = FOOTSTEP_SAND
+	clawfootstep = FOOTSTEP_SAND
+	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	tiled_dirt = FALSE
+	var/ore_type = /obj/item/stack/ore/glass
+	var/turfverb = "dig up"
+	baseturfs = /turf/open/floor/plating
+
+/turf/open/floor/ship/dirt/attackby(obj/item/C, mob/user, params)
+	if((C.tool_behaviour == TOOL_SHOVEL) && params)
+		new ore_type(src, 2)
+		user.visible_message("<span class='notice'>[user] digs up [src].</span>", "<span class='notice'>You [turfverb] [src].</span>")
+		playsound(src, 'sound/effects/shovel_dig.ogg', 50, TRUE)
+		make_plating()
+	if(..())
+		return
+
+/turf/open/floor/ship/dirt/dark
+	icon_state = "greenerdirt"
+
+/turf/open/floor/grass/ship
+	name = "grass"
+	desc = "A patch of grass."
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_TURF_OPEN, SMOOTH_GROUP_FLOOR_GRASS)
+	canSmoothWith = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_FLOOR_GRASS)
+	layer = HIGH_TURF_LAYER
+	var/smooth_icon = 'icons/turf/floors/grass.dmi'
+	baseturfs = /turf/open/floor/ship/dirt
+
+/turf/open/floor/grass/ship/Initialize(mapload, inherited_virtual_z)
```

Comment:
same thing with the type. also, curious how these are removed?

Reply:
These are removed with a shovel, also they are separate from the normal /grass because those do not smooth, don't have  a shipturf dirt baseturf, and result in a shipturf dirt with shoveled.

Reply:
![image](https://github.com/shiptest-ss13/Shiptest/assets/55075747/1357f7b3-128b-4e98-8614-25a60a29c6ce)